docs/ENCODING_BEST_PRACTICES.md

```
@@ -0,0 +1,301 @@
+import json
+data = {"status": "✅", "message": "中文"}
+json_str = json.dumps(data, ensure_ascii=False)  # Preserve unicode
+```
+
+**DON'T:**
+```python
+# ASCII-only JSON
+json_str = json.dumps(data, ensure_ascii=True)  # ❌ Escapes unicode
+```
+
+## Common Pitfalls
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)